perllib/FixMyStreet/App/Controller/Auth/Social.pm

```diff
@@ -237,6 +238,14 @@ sub oidc_callback: Path('/auth/OIDC') : Args(0) {
     # which is passed to Open311 with reports made by this user.
     my $extra = $c->cobrand->call_hook(oidc_user_extra => $id_token);
 
+    # The OIDC endpoint may require a specific URI to be called to log the user
+    # out when they log out of FMS.
+    if ( my $redirect_uri = $c->cobrand->feature('oidc_login')->{logout_uri} ) {
+        $redirect_uri .= "?post_logout_redirect_uri=";
+        $redirect_uri .= URI::Escape::uri_escape( $c->uri_for('/auth/sign_out') );
+        $c->session->{oauth}{logout_redirect_uri} = $redirect_uri;
+    }
+
```

Comment:
It'd be nice to have a test for this.

Reply:
Agreed, and I did write one but ended up rabbit-holing on some issues with the tests not redirecting properly due to some session weirdness. I'll give it another look.